auctions/english-auction/project/auction-contract/tests/artifacts/NFT/src/main.sw

```diff
@@ -0,0 +1,53 @@
+contract;
+
+use sway_libs::nft::{
+    approve,
+    approved,
+    balance_of,
+    is_approved_for_all,
+    mint,
+    NFT,
+    owner_of,
+    set_approval_for_all,
+    tokens_minted,
+    transfer,
+};
+
+impl NFT for Contract {
+    #[storage(read, write)]
+    fn approve(approved_identity: Option<Identity>, token_id: u64) {
+        approve(approved_identity, token_id);
```

Comment:
I recall having issues with variable namespace collision so I was surprised that this works. @emilyaherbert has been working on recursion and I don't know the state that it is in so this might become an issue at some point.